src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -3652,6 +3653,46 @@ def get_no_uptime_sla(self) -> bool:
 
         return self._get_no_uptime_sla(enable_validation=True)
 
+    def get_defender_config(self) -> Union[TypeVar("ManagedClusterSecurityProfileAzureDefender"), None]:
+        """Obtain the value of defender.
+
+        :return: string or None
+        """
+        disable_defender = self.raw_param.get("disable_defender")
+        if disable_defender:
+            return self.models.ManagedClusterSecurityProfileAzureDefender(enabled=False)
+
+        # read the original value passed by the command
+        enable_defender = self.raw_param.get("enable_defender")
+
+        if not enable_defender:
+            return None
+
+        workspace = ""
+        config_file_path = self.raw_param.get("defender_config")
+        if config_file_path:
+            if not os.path.isfile(config_file_path):
+                raise InvalidArgumentValueError(
+                    "{} is not valid file, or not accessable.".format(
+                        config_file_path
+                    )
+                )
+            defender_config = get_file_json(config_file_path)
+            if "logAnalyticsWorkspaceResourceId" in defender_config:
+                workspace = defender_config["logAnalyticsWorkspaceResourceId"]
+
+        if workspace == "":
+            workspace = ensure_default_log_analytics_workspace_for_monitoring(
```

Comment:
Please replace this with 
```suggestion
            workspace = self.external_functions.ensure_default_log_analytics_workspace_for_monitoring(
```
which would provide a chance for overloading in aks-preview.

Also, if a user enables both monitoring and defender addons, the same workspace would be used. Is this expected? And have you tested that this will not affect the behavior of the original monitoring addon?